avm/ptn/app-service-lza/hosting-environment/tests/e2e/waf-aligned/main.test.bicep

```diff
@@ -0,0 +1,41 @@
+metadata name = 'WAF-aligned'
+metadata description = 'This instance deploys the module with WAF aligned settings.'
+
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'appwaf'
+
+@description('Optional. Test name prefix.')
+param namePrefix string = '#_namePrefix_#'
+
+@description('Optional. The password to leverage for the login.')
+@secure()
+param password string = newGuid()
+
+// ============== //
+// Test Execution //
+// ============== //
+
+module testDeployment '../../../main.bicep' = {
```

Comment:
I guess, to be WAF-aligned this would need at least one more thing, that is the diagnostics deployment. For that you can deploy the required dependencies using our helper template

https://github.com/Azure/bicep-registry-modules/blob/ff8dc79cd4cc2794692fee8d6ec7b4e51b36e572/avm/res/key-vault/vault/tests/e2e/waf-aligned/main.test.bicep#L44-L56

and then just reference its outputs when configuring the diagnostic settings